firmware/application/apps/ui_playlist.cpp

```diff
@@ -198,17 +213,41 @@ void PlaylistView::stop(const bool do_loop) {
     if (is_active()) {
         replay_thread.reset();
     }
-    if (do_loop) {
-        if (playlist_db.size() > 0) {
-            start();
+
+    // TODO: the logic here could be more beautiful but maybe they are all same for compiler anyway....
+    // Notes of the logic here in case if it needed to be changed in the future:
+    // 1. check_loop.value() is for the LOOP checkbox
+    // 2. do_loop is a part of the replay thread, not a user - control thing.
+    // 3. when (total_tracks != track_number) is true, it means that the current track is not the last track.
+    // Thus, (do_loop && (total_tracks != track_number)) is for the case when the start() func were called with true AND not the last track.
+    // Which means it do loop until the last track.
+
+    if (check_loop.value()) {
+        if (do_loop) {
+            if (playlist_db.size() > 0) {
+                start();
+            } else {
+                playlist_db = playlist_masterdb;
+                start();
+            }
         } else {
-            playlist_db = playlist_masterdb;
-            start();
+            radio::set_antenna_bias(false);  // Turn off Bias Tee
+            radio::disable();
+            button_play.set_bitmap(&bitmap_play);
+        }
+    } else if (!check_loop.value()) {
+        if (do_loop && (total_tracks != track_number)) {
+            if (playlist_db.size() > 0) {
+                start();
+            } else {
+                playlist_db = playlist_masterdb;
+                start();
+            }
+        } else {
+            radio::set_antenna_bias(false);  // Turn off Bias Tee
```

Comment:
Why? This shouldn’t be changed by an app.

Reply:
For the original logic in replay app, it were called when a single file replay finished, so it could turn off RF frontend (or keep open when do looping). @jimilinuxguy and I used same code here. It would be nice if someday we take care of this, including all other apps. XD

Reply:
#1056